### ISSUE TITLE
Priority #1

### ISSUE THREAD

**Issue by AlbertoDePena** [AlbertoDePena/brain-function-testing#1]
Research possibility to send parameters required to generate a remote test link to CNSVS in real time and receive a response with a valid URL that the patient can be redirected to for test taking.

Acceptance criteria:

As a patient, from BFT portal, click button that will directly take me to my test (link code embedded in query string).

**Comment by AlbertoDePena** [comment]
@brainfunctiontesting 

According to testing platform, "the data will only be sent once. If your site doesn't capture the data we will not be able to resend it and you will have to hand enter it in your database". That puts a constraint on us to always be up (the process that captures that data). I would like to ask if they have a secure API that exposes non sensitive data.

**Comment by brainfunctiontesting** [comment]
I know that was a concern that Daniel had. It's better for us to be able to
fetch data than having to deal with a bunch of POST requests from them.
Yes, let's ask about a secure API or if it's possible for us to make the
requests on our end.


*Chris*
832-639-0790
chris@brainfunctiontesting.com
www.brainfunctiontesting.com

**Comment by AlbertoDePena** [comment]
@brainfunctiontesting 

Awesome, I already sent him an email asking him a few questions =)

**Comment by AlbertoDePena** [comment]
@brainfunctiontesting 

I am not to exited that their current implementation only pushes data to us (not ideal) but if you want to get the ball rolling, then it is what it is. Now if Alan cannot provide a way for us to (calling their system) get a link for a given URL, whats the next action for us to take?

Hope all is well =)

**Comment by brainfunctiontesting** [comment]
Alberto,

Thanks for your patience so far. With CNSVS taking the app server off the
table, I'd still like to discuss what it looks like for us to get a server
up and running that can accept the domain push data and have some kind of
redundancy protocol in place to account for their single data push model.
If we can begin seeing the data that they push to us, it can help us solve
a couple of logistics issues:

   1. Automating the post-test process because we can "see" when tests have
   been taken
   2. Automating the billing process for valid test results
   3. Automating the report creation and sending process to our practices
   (or posting them to an account page inside the website)

Do you have time this week to discuss how we can begin moving on these
specific items?


*Chris*
832-639-0790
chris@brainfunctiontesting.com
www.brainfunctiontesting.com

**Comment by AlbertoDePena** [comment]
@brainfunctiontesting 

Chris,

We can leverage Azure functions to store the data push. Should be very simple. I just need to double check with Alan to get schema for the payload I ll be receiving. I can create a free Azure account using developer@brainfunctiontesting.com then we can make it a real one later. I can start this process right away, you just tell me if that is what you want me to do.

**Comment by brainfunctiontesting** [comment]
Yes. Let's go ahead and set this up so we can start seeing what the real
time data looks like. The first goals after we get this up and running
would be to a) see if we can automate the report creation and sending
process and b) automating the billing process. Underneath both of these
processes, we would need some kind of API with ActiveCampaign so we can
alert the automated email sequence of a successful test completion and turn
off any reminders or other notifications. Help me get an idea of what we're
discussing cost-wise here.

On a separate note, we've had a few successful conversations with CNSVS
about pushing their app server forward on an earlier timetable. Nothing
tangible from it yet, but I'll keep you posted if so!


*Chris*
832-639-0790
chris@brainfunctiontesting.com
www.brainfunctiontesting.com
